### PR TITLE
added templates for basic/ts/djs

### PR DESCRIPTION
**Pull request by KartikeSingh**

### CODE REVIEW

File: templates/basic/TypeScript/discordjs/package.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "discord-bot-ts",
+  "description": "This project was created with **create-discord-app**!",
+  "version": "1.0.0",
+  "main": "build/index.js",
+  "scripts": {
+    "dev": "nodemon src/index.ts",
+    "build": "set NODE_ENV=build && tsc && node build/index.js"
```
Comment (zyrouge) [inline templates/basic/TypeScript/discordjs/package.json:8]: Use cross-env instead, and set `NODE_ENV` to `production`

File: templates/basic/TypeScript/discordjs/src/classes/client.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+        // Setting the commands than handling the events
+        this.setCommands();
+        this.handleEvents();
+        this.login(token);
+    }
+
+    setCommands() {
+        // Reading the command directory
+        readdirSync(`${process.cwd()}/${this.folder}/commands`).forEach(cat => {
```
Comment (zyrouge) [inline templates/basic/TypeScript/discordjs/src/classes/client.ts]: Avoid thread blocking, use promisify `readdir` or use `fs-extra`

File: templates/basic/TypeScript/discordjs/src/classes/client.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                // Setting the command
+                this.commands.set(command.data.name, command);
+            }
+        })
+    }
+
+    // Handling the events
+    handleEvents() {
+        // Reading the event directory
+        readdirSync(`${process.cwd()}/${this.folder}/events`).forEach(event => {
```
Comment (zyrouge) [inline templates/basic/TypeScript/discordjs/src/classes/client.ts]: Avoid thread blocking processes here aswell

File: templates/basic/TypeScript/discordjs/src/classes/client.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                this.once(eventName, (...args) => Event.execute(this, ...args))
+            } else {
+                this.on(eventName, (...args) => Event.execute(this, ...args))
+            }
+        })
+    }
+}
+
+// Defining types
+type commandExecute = (client: Client, message: CommandInteraction) => any;
```
Comment (zyrouge) [inline templates/basic/TypeScript/discordjs/src/classes/client.ts]: Use pascal case and it `Promise<void>`

File: templates/basic/TypeScript/discordjs/src/classes/client.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+interface command {
```
Comment (zyrouge) [inline templates/basic/TypeScript/discordjs/src/classes/client.ts]: Use pascal case here aswell

File: templates/basic/TypeScript/discordjs/package.json (hunk 1 above)
Comment (zyrouge) [inline templates/basic/TypeScript/discordjs/package.json]: Don't start the process on `build`, only run `tsc` and `cross-env NODE_ENV=production node build/index` as `start` script

**Review by zyrouge — CHANGES_REQUESTED** [review]

File: templates/basic/TypeScript/discordjs/package.json [hunk 6]
```diff
@@ -5,7 +5,8 @@
   "main": "build/index.js",
   "scripts": {
     "dev": "nodemon src/index.ts",
-    "build": "set NODE_ENV=build && tsc && node build/index.js"
+    "build": "tsc",
+    "start": "cross-env NODE_ENV=production && node build/index.js"
```
Comment (zyrouge) [inline templates/basic/TypeScript/discordjs/package.json]: Its `cross-env NODE_ENV=production node build/index.js`

**Review by zyrouge — CHANGES_REQUESTED** [review]

File: templates/basic/TypeScript/discordjs/package.json (hunk 6 above)
Comment (KartikeSingh) [reply templates/basic/TypeScript/discordjs/package.json]: ok

Comment (KartikeSingh) [reply templates/basic/TypeScript/discordjs/package.json]: changes done

**Comment by zyrouge** [comment]
Looks good.

**Pull request merged**